Fatras/include/ActsFatras/Kernel/InteractionList.hpp

```diff
@@ -0,0 +1,344 @@
+  template <size_t kProcess>
+  std::tuple_element_t<kProcess, Processes>& get() {
+    return std::get<kProcess>(m_processes);
+  }
+  /// Access a specific process identified by type.
+  ///
+  /// @warning This function only works if all configured processes have
+  ///   different types.
+  template <typename process_t>
+  process_t& get() {
+    return std::get<process_t>(m_processes);
+  }
+
+  /// Simulate the combined effects from all continuous interactions.
+  ///
+  /// @tparam generator_t must be a RandomNumberEngine
+  /// @param[in]     rng       is the random number generator
+  /// @param[in]     slab      is the passed material
+  /// @param[in,out] particle  is the particle being updated
+  /// @param[out]    generated is the container of generated particles
+  /// @return Break condition, i.e. whether a process stoped the propagation
+  template <typename generator_t>
+  bool runContinuous(generator_t& rng, const Acts::MaterialSlab& slab,
+                     Particle& particle,
+                     std::vector<Particle>& generated) const {
+    return runContinuousImpl(rng, slab, particle, generated,
+                             ContinuousIndices());
+  }
+
+  /// Arm the point-like interactions by generating limits and select processes.
+  ///
+  /// @tparam generator_t must be a RandomNumberEngine
+  /// @param[in] rng      is the random number generator
+  /// @param[in] particle is the initial particle state
+  /// @return X0/L0 limits for the particle and the process index that should be
+  ///   executed once the limit has been reached.
+  template <typename generator_t>
+  Selection armPointLike(generator_t& rng, const Particle& particle) const {
+    Selection selection;
+    armPointLikeImpl(rng, particle, selection, PointLikeIndices());
+    return selection;
+  }
+
+  /// Simulate the effects from a single point-like interaction.
+  ///
+  /// @tparam generator_t must be a RandomNumberEngine
+  /// @param[in]     rng          is the random number generator
+  /// @param[in]     processIndex is the index of the process to be executed
+  /// @param[in,out] particle     is the particle being updated
+  /// @param[out]    generated    is the container of generated particles
+  /// @return Break condition, i.e. whether a process killed the particle
+  ///
+  /// The process index is expected to originate from a previous
+  /// `armPointLike(...)` call, but this is not enforced. How to select the
+  /// correct process requires more information that is not available here.
+  template <typename generator_t>
+  bool runPointLike(generator_t& rng, size_t processIndex, Particle& particle,
+                    std::vector<Particle>& generated) const {
+    return runPointLikeImpl(rng, processIndex, particle, generated,
+                            PointLikeIndices());
+  }
+
+ private:
+  // allow processes to be masked. defaults to zeros -> no masked processes
+  Mask m_mask;
```

Comment:
Hm, I see. Conceptually, the other way round would be more intuitive, but it's probably not important.

Reply:
This is the way it was done before. I would prefer to leave it as-is.